File: Array/find_missing.go
package main

import "fmt"

func findMissing(arr []int) int {
	sum := 0
	largest := arr[0]

	for _, val := range arr {
		if largest < val {
			largest = val
		}
		sum += val
	}

	return (largest * (largest + 1) / 2) - sum
}

func main() {
	fmt.Println("[Find Missing Number]\nTime Comp: O(N)")
	arr := []int{1, 3, 7, 5, 6, 2}
	missing := findMissing(arr)
	fmt.Println("The missing number is", missing)
}

File: Array/problems/easy/find_missing.cpp
#include <iostream>

int find_missing(int arr[], int size){
  int largest = arr[0];
  int sum = 0;
  for(int i = 0; i < size; i++){
    if (largest < arr[i])
      largest = arr[i];
    sum += arr[i];
  }

  return (largest * (largest+1))/2 - sum;

}

int main(){
  printf("[Find missing in Array]\nTime Comp: O(N)\n");
  int arr[] = { 1, 3, 7, 5, 6, 2 };
  int size = sizeof(arr)/sizeof(arr[0]);
  int missing = find_missing(arr, size);
  printf("The missing number is: %d\n", missing);

}
